File: area.ts
import request from './request'

export interface Area {
  id?: number
  name: string
  description?: string
  status?: string
  createTime?: string
}

export const areaApi = {
  getList() {
    return request.get<Area[]>('/area')
  },
  getById(id: number) {
    return request.get<Area>(`/area/${id}`)
  },
  create(data: Area) {
    return request.post('/area', data)
  },
  update(data: Area) {
    return request.put(`/area/${data.id}`, data)
  },
  delete(id: number) {
    return request.delete(`/area/${id}`)
  },
}

File: Area.java
package com.example.studyroomserver.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

@Data
@TableName("area")
public class Area {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String description;
    private String status;   // ENABLE / DISABLE
    @TableLogic
    private Integer deleted;
}